### PR TITLE
Windows C# distribtests

### PR DESCRIPTION
**Pull request by jtattermusch**
Add C# distribtest for windows (both 32bit and 64bit).

### CODE REVIEW

File: tools/run_tests/distribtest_targets.py [hunk 1]
```diff
@@ -97,7 +97,14 @@ def build_jobspec(self):
           ['test/distrib/csharp/run_distrib_test.sh'],
           environ={'EXTERNAL_GIT_ROOT': '../../..'})
     else:
```
Comment (nathanielmanistaatgoogle) [inline tools/run_tests/distribtest_targets.py]: Should this not be `elif self.platform == 'windows':`, with the `else:` that raises an exception retained?

File: test/distrib/csharp/run_distrib_test.bat [hunk 2]
```diff
@@ -0,0 +1,21 @@
+
```
Comment (nathanielmanistaatgoogle) [inline test/distrib/csharp/run_distrib_test.bat]: What's with a blank line at top-of-file? Should this file not have a license header like almost every other in the codebase?

Comment (ctiller) [inline test/distrib/csharp/run_distrib_test.bat]: We may want to extend the copyright checker to recognize batch files.
Actually we probably want a whitelist of extensions that don't need
copyrights in the copyright checker (eg vcxproj, sln)

File: tools/run_tests/distribtest_targets.py (hunk 1 above)
Comment (jtattermusch) [inline tools/run_tests/distribtest_targets.py]: I don't see much value in doing that (considering we know we currently support only linux mac an windows, and in this form the script will surely fail if not run on windows). I'm fine with making the change if you insist though.

File: test/distrib/csharp/run_distrib_test.bat (hunk 2 above)
Comment (jtattermusch) [inline test/distrib/csharp/run_distrib_test.bat]: I see, that would mean more work, so let's fix the copyrights for .bat files in a separate PR?

File: tools/run_tests/distribtest_targets.py (hunk 1 above)
Comment (nathanielmanistaatgoogle) [inline tools/run_tests/distribtest_targets.py]: Please.

**Comment by jtattermusch** [comment]
Filed an issue for missing copyrights in .bat files https://github.com/grpc/grpc/issues/5479.

File: test/distrib/csharp/run_distrib_test.bat (hunk 2 above)
Comment (jtattermusch) [inline test/distrib/csharp/run_distrib_test.bat]: Added a copyright notice for this file.

File: tools/run_tests/distribtest_targets.py (hunk 1 above)
Comment (jtattermusch) [inline tools/run_tests/distribtest_targets.py]: Done.

**Comment by nathanielmanistaatgoogle** [comment]
Change content looks good; squash commits if you like and merge when you please.

**Comment by jtattermusch** [comment]
Sanity test is failing:

```
22,29d21
< @rem Collect protoc artifacts built by the previous build step
< xcopy /Y /I ..\..\architecture=x86,language=protoc,platform=windows\artifacts\* protoc_plugins\windows_x86\
< xcopy /Y /I ..\..\architecture=x64,language=protoc,platform=windows\artifacts\* protoc_plugins\windows_x64\
< xcopy /Y /I ..\..\architecture=x86,language=protoc,platform=linux\artifacts\* protoc_plugins\linux_x86\
< xcopy /Y /I ..\..\architecture=x64,language=protoc,platform=linux\artifacts\* protoc_plugins\linux_x64\
< xcopy /Y /I ..\..\architecture=x86,language=protoc,platform=macos\artifacts\* protoc_plugins\macosx_x86\
< xcopy /Y /I ..\..\architecture=x64,language=protoc,platform=macos\artifacts\* protoc_plugins\macosx_x64\
< 
38c30
< @rem We won't use the native libraries from this step, but without this Grpc.sln will fail.
---
> @rem We won't use the native libraries from this step, but without this Grpc.sln will fail.  
44a37,39
> @rem TODO(jtattermusch): re-enable protoc plugin building
> @rem @call ..\..\vsprojects\build_plugins.bat || goto :error
> 
50c45,47
< %NUGET% pack Grpc.Tools.nuspec -Version %VERSION% || goto :error
---
> 
> @rem TODO(jtattermusch): re-enable building Grpc.Tools package
> @rem %NUGET% pack Grpc.Tools.nuspec -Version %VERSION% || goto :error
PASSED: ./grpc.gemspec [time=0.2sec; retries=0:0]
```

I will fix that and then proceed to merging.

**Comment by jtattermusch** [comment]
Ok, so the sanity test failure is actually caused by a different PR. Will fix that separately.

**Pull request merged**